src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,45 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.events.WebDriverEventListener;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+    protected static WebDriver driver;
+    protected static WebDriverWait wait;
+    protected  static Actions action;
+    private static final int time = 6; //time in seconds for timeout wait
+
+    public BasePage (WebDriver submittedDriver) {
+        driver = submittedDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(time));
+        action = new Actions(driver);
+    }
+
+    public static void navigateToPage(String url) {
+        driver.get(url);
+    }
+
+    protected static WebElement findElement (By elementLocator) {
```

Comment:
**protected** means - 'package-private + visibility to subclasses - the property/behaviour is **INHERITED**
**static** means - 'the opposite of instance - it is a CLASS property/behaviour, i.e it is NOT **INHERITED**

Therefore they are slightly contradictive and incompatible.

Hint
![image](https://user-images.githubusercontent.com/104761482/224770046-d27ebba5-e8d3-4441-b5c4-60d4b13c33b2.png)

Reply:
I'm sorry, I don't understand how those two completely different things are incompatible. The tasks here to always use helper methods and everywhere we use static methods so that we don't have to create an instance of a class. I don't know why we do it all the time and don't use in our own code more times just a more usual. 

Private, protected and public mean different visibility (and therefore usage) levels. Every time I see lecturers using public everywhere and anywhere without any given reason, I just cringe inside, as I have been trained for years and years not to use public unless I can give a very logical explanation behind my reasoning why it shouldn't be private or protected at least. Here I used protected as I use this method inside in two other child classes many times and expect to use it in a few more child classes, so I need it to be inherited. 

Also, that's the first time ever I hear that static is not inherited, how is it so, how does it reflect on an ability to be inherited or not? Could you explain please?

Reply:
> I'm sorry, I don't understand how those two completely different things are incompatible. The tasks here to always use helper methods and everywhere we use static methods so that we don't have to create an instance of a class. I don't know why we do it all the time and don't use in our own code more times just a more usual.
> 
> Private, protected and public mean different visibility (and therefore usage) levels. Every time I see lecturers using public everywhere and anywhere without any given reason, I just cringe inside, as I have been trained for years and years not to use public unless I can give a very logical explanation behind my reasoning why it shouldn't be private or protected at least. Here I used protected as I use this method inside in two other child classes many times and expect to use it in a few more child classes, so I need it to be inherited.
> 
> Also, that's the first time ever I hear that static is not inherited, how is it so, how does it reflect on an ability to be inherited or not? Could you explain please?

Thanks!

1.	If a method is used only by other methods of that class, it should be declared as **private**.
2.	The modifier protected can be used for a class member: a method, constructor, field, or nested class/interface. This means that the class member may be accessed only in the body of that same class, a class from the same package, or a successor class, even if it is located in a different package.
3.	The modifier static is used to create methods and variables (fields) belonging to a class and not to an object.
4.	It's useful to know that static methods are linked at compile time, unlike the linking of virtual or non-static methods, which are linked at run time when called on a real object. Accordingly, static methods cannot be overridden in Java, since polymorphism does not apply to them at run time. This is an important limitation to consider when declaring a method static.

Static members are not inherited, and protected members are only visible to subclasses (and of course the containing class), so a protected static has the same visibility as static - **anti-OO in OOPS.**

BUT, You might wonder "what's the reason for a static protected method, considering that it's impossible to call it from anywhere except from an instance of that class?". Usually there isn't much reason to declare a private or protected method as static. 

**Static methods are public or private** (If a method is used only by other methods of that class, it should be declared as private)
.

**Correct solution: public static**